src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -1551,6 +1566,23 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
     return ret;
   }
 
+  // Function to convert a set (ordered or unordered) to a concatenated string
+  template <typename SetType>
+  std::string setToString(const SetType& set, const std::string& separator = ",")
```

Comment:
Just call it `to_string` for simplicity.
```suggestion
  std::string to_string(const SetType& set, const std::string& separator = ",")
```

Reply:
In addition, I don't see this function is used with both type of set (order + unorder) so for simplicity, it is better to just declare the exact type instead of template:
```
std::string to_string(std::set<....> const& set, ...);
```